src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    const { type } = action;
+    let nextState;
+
+    switch (type) {
+      case 'addProperties':
+        const { extraData } = action;
```

Comment:
The variable 'extraData' is declared inside a switch case without block braces, which might lead to unexpected behavior as the variable could be accessible in other cases too. It's better to always use block braces in switch cases.